src/main/java/net/neoforged/neoforge/network/registration/PayloadRegistrar.java

```diff
@@ -41,78 +42,132 @@ private PayloadRegistrar(PayloadRegistrar source) {
+        register(type, codec, serverHandler, clientHandler, List.of(ConnectionProtocol.CONFIGURATION), Optional.empty(), version, optional);
         return this;
     }
 
     /**
      * Registers a client-bound payload for all phases.
      */
-    public <T extends CustomPacketPayload> PayloadRegistrar commonToClient(CustomPacketPayload.Type<T> type, StreamCodec<? super FriendlyByteBuf, T> reader, IPayloadHandler<T> handler) {
-        register(type, reader, handler, List.of(ConnectionProtocol.PLAY, ConnectionProtocol.CONFIGURATION), Optional.of(PacketFlow.CLIENTBOUND), version, optional);
+    public <T extends CustomPacketPayload> PayloadRegistrar commonToClient(CustomPacketPayload.Type<T> type, StreamCodec<? super FriendlyByteBuf, T> codec, IPayloadHandler<T> handler) {
+        register(type, codec, null, handler, List.of(ConnectionProtocol.PLAY, ConnectionProtocol.CONFIGURATION), Optional.of(PacketFlow.CLIENTBOUND), version, optional);
+        return this;
+    }
+
+    /**
+     * Registers a client-bound payload for all phases without a handler. The missing handler must be registered via {@code RegisterClientPayloadHandlersEvent}.
+     */
+    public <T extends CustomPacketPayload> PayloadRegistrar commonToClient(CustomPacketPayload.Type<T> type, StreamCodec<? super FriendlyByteBuf, T> codec) {
+        register(type, codec, null, null, List.of(ConnectionProtocol.PLAY, ConnectionProtocol.CONFIGURATION), Optional.of(PacketFlow.CLIENTBOUND), version, optional);
         return this;
     }
 
     /**
      * Registers a server-bound payload for all phases.
      */
-    public <T extends CustomPacketPayload> PayloadRegistrar commonToServer(CustomPacketPayload.Type<T> type, StreamCodec<? super FriendlyByteBuf, T> reader, IPayloadHandler<T> handler) {
-        register(type, reader, handler, List.of(ConnectionProtocol.PLAY, ConnectionProtocol.CONFIGURATION), Optional.of(PacketFlow.SERVERBOUND), version, optional);
+    public <T extends CustomPacketPayload> PayloadRegistrar commonToServer(CustomPacketPayload.Type<T> type, StreamCodec<? super FriendlyByteBuf, T> codec, IPayloadHandler<T> handler) {
+        register(type, codec, handler, null, List.of(ConnectionProtocol.PLAY, ConnectionProtocol.CONFIGURATION), Optional.of(PacketFlow.SERVERBOUND), version, optional);
         return this;
     }
 
     /**
      * Registers a bidirectional payload for all phases.
      * <p>
      * Consider using {@link DirectionalPayloadHandler} to wrap client and server handlers.
+     *
+     * @deprecated Use {@link #commonBidirectional(CustomPacketPayload.Type, StreamCodec, IPayloadHandler, IPayloadHandler)} instead
+     */
+    @Deprecated
+    public <T extends CustomPacketPayload> PayloadRegistrar commonBidirectional(CustomPacketPayload.Type<T> type, StreamCodec<? super FriendlyByteBuf, T> codec, IPayloadHandler<T> handler) {
```

Comment:
I'm not so sure about suggesting that most people provide a `null` parameter for the client handler. I would have a method that registers just a server handler and expects the client handler to be registered via the event but I can see how that can cause confusion with this old method that registers the same handler for both, so maybe it's not a great idea

Reply:
I agree that passing `null` is not very nice. Is there a practical use case for a bidirectional packet that needs separate handlers on the client and on the server? I would assume that it could just be refactored to be two separate packets. Hence we could keep the bidirectional method that registers the same handler for each direction.

Reply:
My suggestion was to keep `DirectionalPayloadHandler` and keep the single handler arg bidirectional overloads. However I really like the prospect of removing `DirectionalPayloadHandler`.

Reply:
> Is there a practical use case for a bidirectional packet that needs separate handlers on the client and on the server?

Neo's singular bidirectional payload has completely different handlers for server and client